wipp-backend-data/src/main/java/gov/nist/itl/ssd/wipp/backend/data/pyramidannotation/timeslices/PyramidAnnotationTimeSliceController.java

```diff
@@ -0,0 +1,230 @@
+		PyramidAnnotationTimeSlice timeSlice = new PyramidAnnotationTimeSlice(timeSliceId);
+		PyramidAnnotation pyramidAnnotation = pyramidAnnotationRepository.findById(pyramidAnnotationId).get();
+		File pyramidAnnotationFolder = new File(config.getPyramidAnnotationsFolder(), pyramidAnnotationId);
+
+		if(!pyramidAnnotationFolder.exists()) {
+			pyramidAnnotationFolder.mkdirs();
+		}
+
+		List<PyramidAnnotationTimeSlice> allTimeSlices = pyramidAnnotation.getTimeSlices();
+
+		if(containsTimeSlice(allTimeSlices, timeSliceId)) {
+			for (int i = 0; i < allTimeSlices.size(); i++) {
+				if(allTimeSlices.get(i).getSliceNumber() == timeSliceId) {
+					allTimeSlices.set(i, timeSlice);
+				}
+			}
+		} else {
+			allTimeSlices.add(timeSlice);
+		}
+
+		pyramidAnnotation.setTimeSlices(allTimeSlices);
+		pyramidAnnotationRepository.save(pyramidAnnotation);
+		file.transferTo(new File(pyramidAnnotationFolder,
+				PyramidAnnotationConfig.PYRAMID_ANNOTATION_FILENAME_PREFIX + timeSliceId
+				+ PyramidAnnotationConfig.PYRAMID_ANNOTATION_FILENAME_SUFFIX));
+		return pyramidAnnotation;
+	}
+
+
+	private void processResource(String pyramidAnnotationId,
+			PyramidAnnotationTimeSlice pats) {
+		// Self
+		LinkBuilder lb = entityLinks.linkFor(PyramidAnnotationTimeSlice.class,
+				pyramidAnnotationId);
+		Link link = lb.slash(pats.getSliceNumber()).withSelfRel();
+		pats.add(link);
+
+		// Annotation positions
+		lb = entityLinks.linkFor(PyramidAnnotationTimeSlice.class,
+				pyramidAnnotationId);
+		link = lb.slash(pats.getSliceNumber()).slash("annotationPositions")
+				.withRel("annotationPositions");
+		pats.add(link);
+	}
+
+	private Page<PyramidAnnotationTimeSlice> getPage(
+			List<PyramidAnnotationTimeSlice> timeSlices, Pageable pageable) {
+		if (timeSlices == null) {
+			return new PageImpl<>(new ArrayList<>(0), pageable, 0);
+		}
+		long offset = pageable.getOffset();
+		if (offset >= timeSlices.size()) {
+			return new PageImpl<>(
+					new ArrayList<>(0), pageable, timeSlices.size());
+		}
+
+		Stream<PyramidAnnotationTimeSlice> stream = timeSlices.stream();
+		Sort sort = pageable.getSort();
+		Comparator<PyramidAnnotationTimeSlice> comparator
+		= (pts1, pts2) -> Integer.compare(
+				pts1.getSliceNumber(), pts2.getSliceNumber());
+		if (sort != null) {
+			for (Sort.Order order : sort) {
+				if ("sliceNumber".equals(order.getProperty())
+						&& order.getDirection() == Sort.Direction.DESC) {
+					comparator = comparator.reversed();
+				}
+			}
+		}
+		List<PyramidAnnotationTimeSlice> result = stream
+				.sorted(comparator)
+				.skip(offset)
+				.limit(pageable.getPageSize())
+				.collect(Collectors.toList());
+		return new PageImpl<>(result, pageable, timeSlices.size());
+	}
+
+	public boolean containsTimeSlice(List<PyramidAnnotationTimeSlice> timeSlices, int timeSliceId){
+		for(int i = 0; i < timeSlices.size(); i++){
+			if(timeSlices.get(i).getSliceNumber() == timeSliceId){
+				return true;
+			}
+		}
+		return false;
+	}
```

Comment:
Probably not needed, see previous comment

Reply:
removed